test/controllers/journeys/income/IncomeNotCountedAsTurnoverControllerSpec.scala

```diff
@@ -40,122 +41,167 @@ class IncomeNotCountedAsTurnoverControllerSpec extends SpecBase with MockitoSuga
 
-        status(result) mustEqual SEE_OTHER
-        redirectLocation(result).value mustEqual onwardRoute.url
+              status(result) mustEqual OK
+              contentAsString(langResult) mustEqual expectedResult
+            }
+          }
+        }
       }
     }
 
-    "must return a Bad Request and errors when invalid data is submitted" in {
+    "onSubmit" - {
 
-      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+      "must redirect to the next page when valid data is submitted" in {
 
-      running(application) {
-        val request =
-          FakeRequest(POST, incomeNotCountedAsTurnoverRoute)
-            .withFormUrlEncodedBody(("value", ""))
+        val mockSessionRepository = mock[SessionRepository]
 
-        val boundForm = form.bind(Map("value" -> ""))
+        when(mockSessionRepository.set(any())) thenReturn Future.successful(true)
 
-        val view = application.injector.instanceOf[IncomeNotCountedAsTurnoverView]
+        val application =
+          applicationBuilder(userAnswers = Some(emptyUserAnswers))
+            .overrides(
+              bind[Navigator].toInstance(new FakeNavigator(onwardRoute)),
+              bind[SessionRepository].toInstance(mockSessionRepository)
+            )
+            .build()
 
-        val result = route(application, request).value
+        running(application) {
+          val request =
+            FakeRequest(POST, incomeNotCountedAsTurnoverRoute(true, NormalMode)).withFormUrlEncodedBody(("value", "true"))
 
-        status(result) mustEqual BAD_REQUEST
-        contentAsString(result) mustEqual view(boundForm, NormalMode, taxYear)(request, messages(application)).toString
+          val result = route(application, request).value
+
+          status(result) mustEqual SEE_OTHER
+          redirectLocation(result).value mustEqual onwardRoute.url
+        }
       }
-    }
 
-    "must redirect to Journey Recovery for a GET if no existing data is found" ignore { //TODO unignore when RequireData is implemented
+      userScenarios.foreach { userScenario =>
+        s"when language is ${isWelshToString(userScenario.isWelsh)} and user is an ${isAgentToString(userScenario.isAgent)}" - {
+          "must return a Bad Request and errors when an empty form is submitted" in {
 
-      val application = applicationBuilder(userAnswers = None).build()
+            val application          = applicationBuilder(userAnswers = Some(emptyUserAnswers), isAgent = userScenario.isAgent).build()
+            implicit val messagesApi = application.injector.instanceOf[MessagesApi]
 
-      running(application) {
-        val request = FakeRequest(GET, incomeNotCountedAsTurnoverRoute)
+            running(application) {
+              val request =
+                FakeRequest(POST, incomeNotCountedAsTurnoverRoute(true, NormalMode))
+                  .withFormUrlEncodedBody(("value", ""))
+                  .withFormUrlEncodedBody(("value", ""))
```

Comment:
Why do we have two bodies provided here (+ for the test below)?

Reply:
a replacement error, thank you. Removed